flocker/ca/_ca.py

```diff
@@ -231,14 +231,81 @@ def write_credential_files(self, key_filename, certificate_filename):
             os.umask(original_umask)
 
 
-class NodeCredential(FlockerCredential):
+class UserCredential(PRecord):
+    """
+    A certificate for an API user, signed by a supplied certificate
+    authority.
+
+    :ivar FlockerCredential credential: The certificate and key pair
+        credential object.
+    :ivar bytes username: A username.
+    """
+    credential = field(mandatory=True)
+    username = field(mandatory=True, initial=None)
+
+    @classmethod
+    def from_path(cls, path, username):
+        """
+        Load a node certificate from a specified path.
+        """
+        key_filename = b"{user}.key".format(user=username)
+        cert_filename = b"{user}.crt".format(user=username)
+        keypair, certificate = load_certificate_from_path(
+            path, key_filename, cert_filename
+        )
+        credential = FlockerCredential(
+            path=path, keypair=keypair, certificate=certificate)
+        return cls(credential=credential, username=username)
+
+    @classmethod
+    def initialize(cls, path, authority, username):
+        """
+        Generate a certificate signed by the supplied root certificate.
+
+        :param FilePath path: Directory where the certificate will be stored.
+        :param CertificateAuthority authority: The certificate authority with
+            which this certificate will be signed.
+        :param bytes username: The username to be included in the certificate.
```

Comment:
:question: 

Using `bytes` to represent a username means the encoding is up to the caller.  It would be nice to enforce UTF-8 here.